### PR TITLE
Add additional helm chart and yaml linting/verifying

### PR DESCRIPTION
**Pull request by ivanELEC**
## Description of change

[Last outcome for this ticket](https://dsdmoj.atlassian.net/browse/CRM457-1547)

Note: this runs helm lint AND helm template (debug) because from my experience helm lint won't always necessarily catch issues that can break a chart (I tested helm lint on some misconfigurations and it didn't catch some of them) where helm template will AND helm template doesn't give us warning level suggestions like helm lint does - so both in tandem covers all bases for breaking changes and suggestions

Also including yamllint to check for things like duplicate keys upon suggestion from @elken[ in this PR](https://github.com/ministryofjustice/laa-submit-crime-forms/pull/1013)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -128,6 +136,26 @@ commands:
...
+  lint-and-verify-helm-config:
+      description:
+      parameters:
+        environment:
+          description: Name of environment to check helm template of
+          type: string
+      steps:
+        - helm/install_helm_client:
+            version: 3.7.2
+        - *install-yamllint
```
Comment (jsugarman) [inline .circleci/config.yml:151]: These steps only need doing once. Here they would be installed for each environment run, slowing things down slightly.

**Review by jsugarman — COMMENTED** [review]
Personally I am not really sure we need this as part of CI and the cost may outweigh the benefit in this regard. Its only ever something i have done locally while working on helm config and would expected deployments to fail if something was wrong. Is there a risk of false negatives? 

The second command using yamllint gives me lots of errors locally related to indentation, line length and trailing whitespaces but none on CI. Is it supposed to fail on finding linting problem?

**Comment by elken** [comment]
> Personally I am not really sure we need this as part of CI and the cost may outweigh the benefit in this regard. Its only ever something i have done locally while working on helm config and would expected deployments to fail if something was wrong. Is there a risk of false negatives?
> 
> The second command using yamllint gives me lots of errors locally related to indentation, line length and trailing whitespaces but none on CI. Is it supposed to fail on finding linting problem?

The branch includes a config file to ignore the majority of useless errors, it fails for things like duplicate keys that are actually issues but otherwise reports nothing

**Comment by ivanELEC** [comment]
> > Personally I am not really sure we need this as part of CI and the cost may outweigh the benefit in this regard. Its only ever something i have done locally while working on helm config and would expected deployments to fail if something was wrong.
> 
> You would expect it, but things like the `securityContext` being duplicated and just silently applied slipped through without issues. It'll prevent the potentially breaking deployment from going out, it's a safety net.
> 
> > Is there a risk of false negatives?
> 
> No tool is perfect, but it's preferential to the alternative.
> 
> > The second command using yamllint gives me lots of errors locally related to indentation, line length and trailing whitespaces but none on CI. Is it supposed to fail on finding linting problem?
> 
> The branch includes a config file to ignore the majority of useless errors, it fails for things like duplicate keys that are actually issues but otherwise reports nothing

if there are false negatives we can still disable rules like we do with Rubocop etc

**Comment by jsugarman** [comment]
> It'll prevent the potentially breaking deployment from going out,

A truly broken deployment will fail and no pods be replaced in any event i believe, but it depends on the nature of the breakage. The last thing that broke a deployment was an `nindent` with too small a number. These steps do not catch that error when i tried it locally.

I guess my main concern is whether the undoubted benefit of extra checks actually unnecessarily blocks a deployment, not too mentioned requiring more knowledge and maintenance.

Anyway, do not let this stop you putting it in as we can trial it regardless. I think it is worth separating the install step from the command if possible though

**Comment by elken** [comment]
> > It'll prevent the potentially breaking deployment from going out,
> 
> A truly broken deployment will fail and no pods be replaced in any event i believe, but it depends on the nature of the breakage. The last thing that broke a deployment was an `nindent` with too small a number. These steps do not catch that error when i tried it locally.

Indentation is one of the disabled rules, if it's one we want to catch then we can add it in. There are a few options available for it: https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.indentation

**Review by jsugarman — APPROVED** [review]

**Pull request merged**